sdk/pubkey/src/lib.rs

```diff
@@ -1114,9 +1114,50 @@ macro_rules! pubkey {
     };
 }
 
+/// New random Pubkey for tests and benchmarks.
+#[cfg(all(feature = "rand", not(target_os = "solana")))]
+pub fn new_rand() -> Pubkey {
+    Pubkey::from(rand::random::<[u8; PUBKEY_BYTES]>())
+}
```

Comment:
I'm a bit torn about this one, since we should really favor using `Pubkey::new_unique()` in almost all cases. On the flipside, there are some legit usecases, and this is the best place to put it, so let's go with it.